### PR TITLE
active and disabled sections in messages connections list

### PR DESCRIPTION
**Pull request by wittejm**
https://user-images.githubusercontent.com/2104990/146834470-81651e8f-d6bb-44fc-9b1f-fb65317bc734.mov

Connections with "pending" or "rejected" status are not shown in messages view. 

There are some styling mismatches from figma. Ticket for hi-fi changes: Ticket: https://app.shortcut.com/ameelio/story/2015/hifi-styling-for-messages-connection-list-active-disabled

Functionally, "inactive" connection status has no impact on ability to view/send messages. Ticket: https://app.shortcut.com/ameelio/story/2013/connection-status-and-connection-restrictions-should-restrict-ability-to-read-send-messages-pathways

### CODE REVIEW

File: src/components/Inbox/ConnectionListSidebar/ConnectionListItem.tsx [hunk 1]
```diff
@@ -15,31 +15,40 @@ export type TConnectionListItem = ConnectionFragmentFragment & {
 interface Props {
   connection: TConnectionListItem;
   lastMessage?: string;
+  selected: boolean;
```
Comment (gabrielsaruhashi) [inline src/components/Inbox/ConnectionListSidebar/ConnectionListItem.tsx:18]: nit: selected?: boolean

File: src/components/Inbox/ConnectionListSidebar/ConnectionListItem.tsx [hunk 2]
```diff
@@ -15,31 +15,40 @@ export type TConnectionListItem = ConnectionFragmentFragment & {
...
-            {getFullName(connection.visitor)}
-          </Typography.Text>
-          {connection.messages.unreadMessageCount > 0 && (
-            <Typography.Text className="ant-typography text-base text-blue-500 pl-2">
-              {` ${connection.messages.unreadMessageCount} new`}
+    <Row
+      className={`border-t ${selected ? "bg-blue-50" : ""
+        } `}
+      key={connection.id}
+      onClick={() => onSelectConnection()}
```
Comment (gabrielsaruhashi) [inline src/components/Inbox/ConnectionListSidebar/ConnectionListItem.tsx:29]: nit 

onClick={onSelectConnection}

File: src/components/Inbox/ConnectionListSidebar/ConnectionList.tsx [hunk 3]
```diff
@@ -15,23 +17,35 @@ const ConnectionList: React.FC<Props> = ({
   onSelect,
   className,
 }) => {
+
+  const groupedConnections = groupBy(
+    connections,
+    (conn) => conn.status,
+    [ConnectionStatus.Active, ConnectionStatus.Inactive, ConnectionStatus.Pending, ConnectionStatus.Rejected]
```
Comment (gabrielsaruhashi) [inline src/components/Inbox/ConnectionListSidebar/ConnectionList.tsx:24]: happy that these default groups were useful / it was clear enough how it worked!

File: src/components/Inbox/ConnectionListSidebar/ConnectionList.tsx [hunk 4]
```diff
@@ -15,23 +17,35 @@ const ConnectionList: React.FC<Props> = ({
...
-        >
-          <InboxChatListItem connection={connection} />
-        </Row>
-      ))}
-    </Layout.Sider>
+      <Collapse defaultActiveKey={["1", "2"]} bordered={false}>
+        <Collapse.Panel header={<span className="pl-4 pt-4 pb-2 text-xs text-gray-400">ACTIVE</span>} key="1" collapsible="disabled">
+          {
+            groupedConnections[ConnectionStatus.Active].map((connection) => (
+              <InboxChatListItem connection={connection} onSelectConnection={() => onSelect(connection)} selected={selected?.id === connection.id} />
```
Comment (gabrielsaruhashi) [inline src/components/Inbox/ConnectionListSidebar/ConnectionList.tsx:36]: you're missing a key here
it'll throw warnigns

File: src/components/Inbox/ConnectionListSidebar/ConnectionList.tsx [hunk 5]
```diff
@@ -15,23 +17,35 @@ const ConnectionList: React.FC<Props> = ({
...
+            ))
+          }
+        </Collapse.Panel>
+        {groupedConnections[ConnectionStatus.Inactive].length > 0 &&
+          <Collapse.Panel style={{ "borderBottom": "0px" }} header={<span className="pl-4 pt-4 pb-2 text-xs text-gray-400">DISABLED</span>} key="2">
+            {groupedConnections[ConnectionStatus.Inactive].map((connection) => (
```
Comment (gabrielsaruhashi) [inline src/components/Inbox/ConnectionListSidebar/ConnectionList.tsx:42]: missing key

File: src/components/Inbox/ConnectionListSidebar/ConnectionList.tsx [hunk 6]
```diff
@@ -15,23 +17,35 @@ const ConnectionList: React.FC<Props> = ({
...
+      <Collapse defaultActiveKey={["1", "2"]} bordered={false}>
+        <Collapse.Panel header={<span className="pl-4 pt-4 pb-2 text-xs text-gray-400">ACTIVE</span>} key="1" collapsible="disabled">
+          {
+            groupedConnections[ConnectionStatus.Active].map((connection) => (
+              <InboxChatListItem connection={connection} onSelectConnection={() => onSelect(connection)} selected={selected?.id === connection.id} />
+            ))
+          }
+        </Collapse.Panel>
+        {groupedConnections[ConnectionStatus.Inactive].length > 0 &&
+          <Collapse.Panel style={{ "borderBottom": "0px" }} header={<span className="pl-4 pt-4 pb-2 text-xs text-gray-400">DISABLED</span>} key="2">
```
Comment (gabrielsaruhashi) [inline src/components/Inbox/ConnectionListSidebar/ConnectionList.tsx:41]: try className="border-b-0"

**Review by gabrielsaruhashi — APPROVED** [review]

File: src/components/Inbox/ConnectionListSidebar/ConnectionList.tsx (hunk 6 above)
Comment (wittejm) [reply src/components/Inbox/ConnectionListSidebar/ConnectionList.tsx:41]: I did and it had no effect :( , looks like the className here maps to a different dom component. 
Just one of many examples I've encountered where antd and tailwind fail to work together as expected.

**Pull request merged**